pkg/server/server.go

```diff
@@ -215,6 +215,8 @@ func (s *Server) HandleSubscribe(c echo.Context) error {
 		go func() {
 			for {
 				lastSeq, err := s.Consumer.ReplayEvents(ctx, sub.compress, *sub.cursor, playbackRateLimit, func(ctx context.Context, timeUS int64, did, collection string, getEventBytes func() []byte) error {
+					sub.lk.Lock()
+					defer sub.lk.Unlock()
 					return emitToSubscriber(ctx, log, sub, timeUS, did, collection, true, getEventBytes)
```

Comment:
hmm. `emitToSubscriber` with `playback = true` would do a blocking send on the outbox while holding the subscriber's lock. could this hold up the Emit semaphore when it tries to acquire the sub's lock in the overlap time of cutover?